tools/jit/gen_jit_dispatch.py

```diff
@@ -162,15 +164,27 @@ def emit_decl_variant(decl, is_positional_arg, has_tensorlist):
             num_dynamic_inputs = static_inputs
 
         real_inputs = count()
+        if has_tensorlist:
+            kw_assignments.append('size_t _Arg_idx = 0;')
         for i, arg in enumerate(decl['arguments']):
             # XXX: we currently support only TensorList ops that have a TensorList as
             # the first argument, that is then followed by a number of positional args.
             if arg['simple_type'] == 'TensorList':
-                arguments.append('peekSlice(stack, 0, varargs_length - {}, varargs_length)'.format(static_inputs))
+                kw_assignments.append('size_t _Arg_idx_{} = _Arg_idx;'.format(i))
+                kw_assignments.append('_Arg_idx += varargs_length - {};'.format(static_inputs))
+                arguments.append(
+                    'peekSlice(stack, _Arg_idx_{}, varargs_length - {}, varargs_length)'.format(i, static_inputs))
             elif arg['simple_type'] in default_only_types:
                 arguments.append(arg['default'])
             elif is_tensor_arg(arg):
-                arguments.append('std::move(peek(stack, {}, {}))'.format(next(real_inputs), static_inputs))
+                if has_tensorlist:
+                    kw_assignments.append('size_t _Arg_idx_{} = _Arg_idx++;'.format(i))
```

Comment:
I am pretty sure there are missing cases for the `is_positional_arg` cases below. Changing all of the sites that use `peek` in this way is pretty fragile.